pytorch_lightning/trainer/trainer.py

```diff
@@ -2616,6 +2617,41 @@ def weights_summary(self, val: Optional[str]) -> None:
     Other
     """
 
+    @property
+    def estimated_num_optimization_steps(self) -> int:
```

Comment:
If this does not account for multiple optimizers or TBPTT splits, then I don't think this is an appropriate name.

I had to make a similar choice in #11805 where I used `batches_that_stepped`. I think we should have a similar name (estimated_stepping_batches?)

Also remember to update the CHANGELOG entry and PR description after we decide!

Reply:
I can't actually find any other name that would come closer, so I think I'm fine with the optimization steps. We can't go for "estimated_batches_steps" or similar because it takes into account accumulation (which in turn is both connected to the number of batches and the optimization steps).  

maybe a warning/info for when multiple optimizers are present?

Reply:
well, when using multiple optimizers, IMO, a warning won't make sense because it will still be optimization steps for each optimizer. I guess the problem here is with the name since by `optimization_steps` do we mean global optimization_steps or total optimization_steps for all the optimizers?

and I can't think of a better name..

Reply:
> We can't go for "estimated_batches_steps" or similar because it takes into account accumulation (which in turn is both connected to the number of batches and the optimization steps).

"batches that stepped" refers to "batches where an optimizer will step". That takes into account accumulation.

If this estimate is called `*_optimization_steps` then it will be confusing after #11805  because it will be different than the global step at the end of training in the case of multiple optimizers or tbptt.

Would you like to discuss this off github?

Reply:
did you guys discuss this offline?

Reply:
@carmocca @awaelchli

Reply:
hey  @kaushikb11 !
we discussed this .. will be updated the PR soon :)